helm/chart-operator-chart/templates/deployment.yaml

```diff
@@ -21,7 +21,20 @@ spec:
         giantswarm.io/service-type: "managed"
       annotations:
         releasetime: {{ $.Release.Time }}
+    tolerations:
+      - key: node-role.kubernetes.io/master
+        effect: NoSchedule
```

Comment:
Can you explain how the tolerations work? Does that mean that not scheduling is tolerated? That would be weird because below is the affinity to put it onto master.

Reply:
A master node has a taint which looks like this (`kubectl describe node`): 
```
Taints:             node-role.kubernetes.io/master:NoSchedule
```
To have pods scheduled on that node you need to match the taint in your toleration. So the code above will make it possible to schedule `chart-operator` to master.

We use the same toleration in our exporters for example:
https://github.com/giantswarm/net-exporter/blob/2cc712a8f482a746ee9192168423ba21661532a8/helm/net-exporter-chart/templates/daemonset.yaml#L46-L48

Reply:
I see. Cool. I think it is the negated toleration that freaks me out.